engine/src/flutter/lib/ui/semantics.dart

```diff
@@ -962,6 +961,233 @@ class SemanticsFlag {
   String toString() => 'SemanticsFlag.$name';
 }
 
+class SemanticsFlags {
+  bool hasCheckedState;
+  bool isChecked;
+  bool isSelected;
+  bool isButton;
+  bool isTextField;
+  bool isFocused;
+  bool hasEnabledState;
+  bool isEnabled;
+  bool isInMutuallyExclusiveGroup;
+  bool isHeader;
+  bool isObscured;
+  bool scopesRoute;
+  bool namesRoute;
+  bool isHidden;
+  bool isImage;
+  bool isLiveRegion;
+  bool hasToggledState;
+  bool isToggled;
+  bool hasImplicitScrolling;
+  bool isMultiline;
+  bool isReadOnly;
+  bool isFocusable;
+  bool isLink;
+  bool isSlider;
+  bool isKeyboardKey;
+  bool isCheckStateMixed;
+  bool hasExpandedState;
+  bool isExpanded;
+  bool hasSelectedState;
+  bool hasRequiredState;
+  bool isRequired;
```

Comment:
Can we make all fields `final`? Otherwise, `operator ==` and `hashCode` would misbehave, and sharing values of this class would be difficult because it's easy to mutate an instance that should not be mutated. If it's convenient to have a mutable variant of this class, consider using the builder-value pattern. The builder - `SemanticsFlagsBuilder` - is mutable, it's not expected to be shared, and it doesn't have `operator==` or `hashCode`. However, the value - `SemanticsFlags` - is immutable, safe to share, and it provides `operator==` and `hashCode`.